### ISSUE TITLE
False positive with JSpecify nullable array in record with a custom constructor

### ISSUE THREAD

**Issue by nvervelle** [uber/NullAway#1631]
Hi, I'm having a strange situation where NullAway incorrectly reports that a `@NonNull` value is required. I've managed to reduce it to the following test class : it fails on the static method, but only if the public custom constructor is defined (if I remove the custom constructor, NullAway reports no problem with the static method)

```java
import java.util.Objects;
import org.jspecify.annotations.Nullable;

public record Test(Object type, String filename, byte @Nullable [] value) {

  public static Test file(final String filename) {
    return new Test(new Object(), filename, null);
  }

  public Test {
    Objects.requireNonNull(type, "type");
    Objects.requireNonNull(filename, "filename");
  }
}

```

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':....:compileJava' (registered by plugin class 'org.gradle.api.plugins.JavaBasePlugin').
> Compilation failed; see the compiler output below.
  /.../Test.java:9: error: [NullAway] passing @Nullable parameter 'null' where @NonNull is required
      return new Test(new Object(), filename, null);
                                              ^
      (see http://t.uber.com/nullaway )
  1 error
```

I'm using NullAway 0.13.7, with errorprone 2.50.0

**Comment by dev-kvt** [comment]
@nvervelle  hi ! 
https://github.com/uber/NullAway/pull/1632#issue-4904753071

**Comment by msridhar** [comment]
Hi @nvervelle I can reproduce this issue but only outside JSpecify mode.  Are you able to run NullAway in JSpecify mode (passing the `JSpecifyMode=true` parameter, [docs](https://github.com/uber/NullAway/wiki/JSpecify-Support))?  It's still a bug we should fix, but if you can run in JSpecify mode that's a good workaround for the moment.

**Comment by msridhar** [comment]
@nvervelle I have a proper fix up for this in #1636.  It should work for you as long as you are compiling with a recent JDK version or with the right flags passed (see [here](https://github.com/uber/NullAway/wiki/JSpecify-Support#supported-jdk-versions)).

**Comment by nvervelle** [comment]
Thanks @msridhar and @dev-kvt .

I'm currently compiling with JDK 25, I will check if I have to change anything with the fix.

**Comment by msridhar** [comment]
@nvervelle I think you shouldn't have to change anything with release 0.13.8 from yesterday.  I do still recommend running in JSpecify mode if you can; eventually that will become the default.

**Comment by nvervelle** [comment]
@msridhar I confirm that I could remove the @SuppressWarnings after upgrading NullAway 👍 

I will check JSpecify mode and report if I see any trouble
